### ISSUE TITLE
[HOLD for payment 07-31-2024][$250] CRITICAL [UX Reliability] The loading skeleton exhibits three different behaviors

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#43791]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.83-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @muttmuure 
**Slack conversation:** https://expensify.slack.com/archives/C05LX9D6E07/p1718365848007159
## Action Performed:
1. Go to staging.new.expensify.com
2. Click on any report in LHN

## Expected Result:
Skeleton animation work as expected
## Actual Result:
At times the static image of the skeleton shows and the skeleton doesn’t appear at all.
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/60111ec1-9434-4e3a-968b-f42e70448721

https://github.com/Expensify/App/assets/38435837/5c7d9535-1d03-43b3-908f-51e522f05689

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013a31dcebc0c8e5de</li>
        <li>Upwork Job ID: 1802755576680221726</li>
        <li>Last Price Increase: 2024-06-17</li>
<li>Automatic offers: </li>
<ul>
<li>akinwale | Reviewer | 102780731</li>
<li>tsa321 | Contributor | 102780734</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @lschurr</details>

**Comment by tsa321** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
UX Reliability issue: The loading skeleton exhibits three different behaviors. Sometimes skeleton view shown above one or two report actions, sometimes doesn't show up, sometimes full cover skeleton shown in

### What is the root cause of that problem?

When user open app / first OpenApp, back end returns certain reportActions types (IOU report actions and some others...). There are three different case: 

1. When opening a report the report actions returned by OpenApp will be displayed at bottom and skeleton shown above it. When OpenReport request completed the report actions moved to proper position. 

2. For other case when full skeleton view is displayed it means, there is no cached report actions (there is no report actions type IOU, or other type that returned by OpenApp). So cached report actions is empty.
 
3. Case when user open report then App immediately displays the messages, it means there are cached report actions in onyx.

### What changes do you think we should make in order to solve the problem?

When opening a report, OpenReport will return around 100 latest reportActions. We could show full skeleton view when getContinous reportActions :

https://github.com/Expensify/App/blob/75c104bfe009ef937bcd129a35bb5a2e52f4339d/src/pages/home/ReportScreen.tsx#L257-L261

When the continuous   reportActions doens't reach 100 or the oldest reportAction type isn't `CREATED` type. 
We should wait for OpenReport to be completed and show the full skeleton view. 

In the end there will be two behavior, If cached report actions exist and fulfill the criteria above we immediately show it. If the above criteria doesn't met we display full skeleton view and wait for OpenReport to be completed.

**Comment by tgolen** [comment]
I'm going to discuss this internally first to see what the expected behavior should be.

**Comment by tgolen** [comment]
OK, it sounds like the solution from @tsa321 is inline with our expectations. I'm going to label this as external and then the C+ can give the proposal a look.

**Comment by muttmuure** [comment]
@abdulrahuman5196 is ooo, let's reapply the label

**Comment by akinwale** [comment]
@tsa321's [proposal](https://github.com/Expensify/App/issues/43791#issuecomment-2169530120) makes sense here.

🎀👀🎀 C+ reviewed.

**Comment by tsa321** [comment]
@akinwale PR is ready...

**Comment by tsa321** [comment]
I am sorry for the regression. I am still working on a new PR for this issue and will submit it later.

**Comment by tsa321** [comment]
@akinwale, my PR is actually ready. However, there is a bug in the main branch related to comment linking. When a user returns from a transaction thread by clicking the link header, the parent report scrolls down but does not focus on the highlighted/linked report action. After merging the latest main into my branch, the bug persists, which will interfere with testing my PR.

Video of the bug:
 
https://github.com/Expensify/App/assets/114975543/58ba2cc6-d7ac-4274-a1c7-abbc0088a4be

**Comment by tsa321** [comment]
Waiting for https://github.com/Expensify/App/issues/44625 to be fixed.

**Comment by lschurr** [comment]
Added hold for #44625. Noting that the payment amounts will be subject to the regression penalty.

**Comment by tsa321** [comment]
@akinwale PR is ready

**Comment by lschurr** [comment]
Just clarifying - What are we waiting on here @akinwale @tsa321? Is this bug now fixed?

**Comment by akinwale** [comment]
@lschurr There was a regression, so I'll need to review the PR.

**Comment by lschurr** [comment]
Any update here @akinwale? Have you had a chance to review the PR?

**Comment by akinwale** [comment]
> Any update here @akinwale? Have you had a chance to review the PR?

Not yet. I will wrap this up tonight. Thanks for the reminder.

**Comment by tsa321** [comment]
Update: The PR was deployed to production 3 days ago.

**Comment by akinwale** [comment]
@lschurr It looks like this is ready for payment as the follow-up PR has been in production for over a week now. Thanks.

**Comment by lschurr** [comment]
Thanks @akinwale - And just to clarify, the original bug here is now fixed, but the first PR caused a regression, correct? https://github.com/Expensify/App/pull/44488 fixed that, right?

**Comment by akinwale** [comment]
> Thanks @akinwale - And just to clarify, the original bug here is now fixed, but the first PR caused a regression, correct? #44488 fixed that, right?

Yes, this is correct.

**Comment by lschurr** [comment]
Payment summary (50% due to regression):

- Contributor: $125 @tsa321 (paid in Upwork)
- Reviewer: $125 @akinwale (paid in Upwork)

**Comment by lschurr** [comment]
Payments are complete.

**Comment by muttmuure** [comment]
I think we needed a regression test for this one

**Comment by muttmuure** [comment]
Do you think we can create a regression test here? @akinwale?

**Comment by akinwale** [comment]
> * [x]  [@akinwale] The PR that introduced the bug has been identified. Link to the PR:
> * [x]  [@akinwale] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
> * [x]  [@akinwale] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

Not a regression.

> * [x]  [@akinwale] Determine if we should create a regression test for this bug.
> * [x]  [@akinwale] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

**Regression Test Steps**
1. Launch Expensify
2. If already logged in, sign out and then log in again
3. Open an IOU request report.
4. Verify that the The IOU preview is not displayed, but the full loading skeleton is displayed until data is received from the server.
5. Open another report and then reopen the initial IOU report.
6. Verify that the initial IOU report opens instantly.
7. Repeat steps 3 through 6 for other kinds of reports.
8. Click the IOU report preview.
9. Verify that the transaction report is opened.
10. Navigate back using the header link from the transaction report using the header link.
11. Verify that navigation occurs correctly back to the report preview.

Do we agree 👍 or 👎?

**Comment by akinwale** [comment]
@muttmuure Done.
